src/models/connect-four-games.ts

```diff
@@ -0,0 +1,210 @@
+
+import { mongoose } from "@services";
+import { BadRequestError } from "@errors";
+
+class ConnectFourGames extends mongoose.Repository<IConnectFour> {
+	constructor() {
+		super("ConnectFourGame", {
+			server_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			red_user_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			yellow_user_id: {
+				type: String,
+				index: true,
+				required: true
+			},
+			to_move: {
+				type: String,
+				required: false,
+				default: undefined
+			},
+			board: {
+				required: false,
+				default: [[], [], [], [], [], [], []],
+				type: [
+					[
+						{
+							type: String,
+							enum: ConnectFourColor
+						}
+					]
+				]
+			},
+			wager: {
+				type: Number,
+				required: false,
+				default: 0
+			},
+			status: {
+				type: String,
+				required: false,
+				default: undefined
+			},
+			is_accepted: {
+				type: Boolean,
+				required: false,
+				default: false
+			},
+			is_complete: {
+				type: Boolean,
+				required: false,
+				default: false
+			}
+		});
+	}
+
+	/** Return user's active game or open/unaccepted challenge if it exists, else null */
+	public async getActiveGame(user: IUser, challenge?: boolean) {
+		const { user_id, server_id } = user;
+		return await super.read({
+			server_id,
+			$or: [{ red_user_id: user_id }, { yellow_user_id: user_id }],
+			is_accepted: challenge ? false : true,
+			is_complete: false
+		});
+	}
+
+	/** Return user's open/unaccepted active challenge if it exists, else null */
+	public async getUnacceptedChallenge(user: IUser) {
+		return await this.getActiveGame(user, true);
+	}
+
+	/** Return true if user has an active game, else false */
+	public async hasActiveGame(user: IUser) {
+		if (await this.getActiveGame(user)) return true;
+		return false;
+	}
+
+	/** Challenge another player to a game */
+	public async createNewChallenge(challengerUser: IUser, challengedUser: IUser, wager: number) {
+		return await super.insertOne({
+			server_id: challengerUser.server_id,
+			red_user_id: challengerUser.user_id,
+			yellow_user_id: challengedUser.user_id,
+			wager
+		});
+	}
+
+	/** Update an existing challenge if necessary (either target a different opponent player, or update the wager amount) */
+	public async updateExistingChallenge(
+		existingChallenge: IConnectFour,
+		challengedUser: IUser,
+		wager: number
+	) {
+		if (
+			existingChallenge.yellow_user_id === challengedUser.user_id &&
+			existingChallenge.wager === wager
+		)
+			return existingChallenge;
+		const { _id } = existingChallenge;
+		return await super.updateOne({ _id }, { yellow_user_id: challengedUser.user_id, wager });
+	}
+
+	/** Randomly pick which player moves first and start the game */
+	public async acceptExistingChallengeAndStartGame(existingChallenge: IConnectFour) {
+		const { _id, red_user_id, yellow_user_id } = existingChallenge;
+		const to_move = Math.random() < 0.5 ? red_user_id : yellow_user_id;
```

Comment:
i think i added the chance library for seeding random values somewhere in lib or helpers